src/main/java/org/opentripplanner/model/modes/AllowedTransitMode.java

```diff
@@ -51,7 +51,12 @@ public static Set<AllowedTransitMode> getAllTransitModesExceptAirplane() {
    * Check if this filter allows the provided TransitMode
    */
   public boolean allows(TransitMode transitMode, String netexSubMode) {
-    return mainMode == transitMode && (subMode == null || subMode.equals(netexSubMode));
+    boolean mainModeMatch = mainMode == transitMode;
+    boolean submodeMatch =
+      subMode == null ||
+      subMode.equals(netexSubMode) ||
+      (subMode.equals("unknown") && netexSubMode == null);
```

Comment:
Can you please move the string "unknown" into a constant?

Reply:
Done